File: app/src/main/java/ie/dublinbuspal/android/di/ApplicationComponent.java
package ie.dublinbuspal.android.di;

import javax.inject.Singleton;

import dagger.Component;
import ie.dublinbuspal.android.view.route.RouteActivity;
import ie.dublinbuspal.android.view.settings.SettingsActivity;

@Singleton
@Component(modules = {ApplicationModule.class, PresenterModule.class, RepositoryModule.class})
public interface ApplicationComponent {

    void inject(RouteActivity routeActivity);

    void inject(SettingsActivity.MainPreferenceFragment mainPreferenceFragment);

}

File: app/src/main/java/ie/dublinbuspal/android/di/ApplicationComponent.kt
package ie.dublinbuspal.android.di

import dagger.Component
import ie.dublinbuspal.android.view.livedata.LiveDataPresenter
import ie.dublinbuspal.android.view.nearby.NearbyPresenter
import ie.dublinbuspal.android.view.routeservice.RouteServicePresenter
import ie.dublinbuspal.android.view.search.SearchPresenter
import ie.dublinbuspal.database.di.DatabaseModule
import ie.dublinbuspal.domain.di.RepositoryModule
import ie.dublinbuspal.service.di.NetworkModule
import javax.inject.Singleton

@Singleton
@Component(
        modules = [
            ApplicationModule::class,
            NetworkModule::class,
            DatabaseModule::class,
            RepositoryModule::class
        ]
)
interface ApplicationComponent {

    fun nearbyPresenter(): NearbyPresenter

    fun liveDataPresenter(): LiveDataPresenter

    fun searchPresenter(): SearchPresenter

    fun routePresenter(): RouteServicePresenter

}
